catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -1124,6 +1133,37 @@ private FilterCapabilities buildFilterCapabilities() {
     return filterCapabilities;
   }
 
+  private List<String> getQueryFilterTransformerTypeNames() {
+    return Stream.of(getBundle())
+        .map(Bundle::getBundleContext)
+        .map(
+            new Function<BundleContext, Collection<ServiceReference<QueryFilterTransformer>>>() {
+              @Override
+              public Collection<ServiceReference<QueryFilterTransformer>> apply(
+                  BundleContext bundleContext) {
+                try {
+                  return bundleContext.getServiceReferences(QueryFilterTransformer.class, null);
+                } catch (InvalidSyntaxException e) {
+                  LOGGER.debug("Unable to get service references for QueryFilterTransformers.", e);
+                }
+                return Collections.emptyList();
+              }
+            })
+        .flatMap(Collection::stream)
+        .map(
+            queryFilterTransformerServiceReference ->
+                queryFilterTransformerServiceReference.getProperty(
+                    QUERY_FILTER_TRANSFORMER_TYPE_NAMES_FIELD))
+        .filter(Objects::nonNull)
+        .filter(List.class::isInstance)
+        .map(List.class::cast)
+        .flatMap((Function<List, Stream<Object>>) Collection::stream)
+        .filter(String.class::isInstance)
+        .map(String.class::cast)
+        .distinct()
```

Comment:
Might alternatively consider returning a set instead of calling distinct

Reply:
Might not be an option if the caller of this method is expecting a List instead of a Collection

Reply:
I can return a set.